### ISSUE TITLE
Unable to compile on Arch Linux

### ISSUE THREAD

**Issue by guihkx** [vrmiguel/bustd#6]
I'm trying to compile bustd on Arch, but I'm getting this:

```
$ cargo build --release
   Compiling proc-macro2 v1.0.27
   Compiling libc v0.2.97
   Compiling unicode-xid v0.2.2
   Compiling syn v1.0.73
   Compiling unicode-segmentation v1.8.0
   Compiling argh_shared v0.1.5
   Compiling cc v1.0.68
   Compiling boxfnonce v0.1.1
   Compiling cfg-if v1.0.0
   Compiling heck v0.3.3
   Compiling quote v1.0.9
   Compiling daemonize v0.4.1
   Compiling bustd v0.1.0 (/home/gui/bustd)
   Compiling argh_derive v0.1.5
   Compiling no-panic v0.1.15
   Compiling argh v0.1.5
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-m64" "/home/gui/bustd/target/release/deps/bustd-29598bf87e19bc74.bustd.4072317c-cgu.0.rcgu.o" "-Wl,--as-needed" "-L" "/home/gui/bustd/target/release/deps" "-L" "/home/gui/bustd/target/release/build/bustd-6bc53eed358101fc/out" "-L" "/usr/lib64/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "-Wl,--whole-archive" "-lhelper" "-Wl,--no-whole-archive" "-Wl,--start-group" "-Wl,--end-group" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-0c7e43ac4e512046.rlib" "-Wl,-Bdynamic" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-znoexecstack" "-L" "/usr/lib64/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/home/gui/bustd/target/release/deps/bustd-29598bf87e19bc74" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-Wl,-O1" "-nodefaultlibs"
  = note: /usr/bin/ld: /home/gui/bustd/target/release/deps/bustd-29598bf87e19bc74.bustd.4072317c-cgu.0.rcgu.o: in function `<bustd::memory::mem_info::sys_info::__NoPanic as core::ops::drop::Drop>::drop':
          bustd.4072317c-cgu.0:(.text._ZN86_$LT$bustd..memory..mem_info..sys_info..__NoPanic$u20$as$u20$core..ops..drop..Drop$GT$4drop17h281314e3600cb451E+0x3): undefined reference to `
          
          ERROR[no-panic]: detected panic in function `sys_info`
          '
          /usr/bin/ld: /home/gui/bustd/target/release/deps/bustd-29598bf87e19bc74.bustd.4072317c-cgu.0.rcgu.o: in function `<bustd::memory::mem_lock::_mlockall_wrapper::__NoPanic as core::ops::drop::Drop>::drop':
          bustd.4072317c-cgu.0:(.text._ZN95_$LT$bustd..memory..mem_lock.._mlockall_wrapper..__NoPanic$u20$as$u20$core..ops..drop..Drop$GT$4drop17h8622ff938e405c5dE+0x3): undefined reference to `
          
          ERROR[no-panic]: detected panic in function `_mlockall_wrapper`
          '
          /usr/bin/ld: /home/gui/bustd/target/release/deps/bustd-29598bf87e19bc74.bustd.4072317c-cgu.0.rcgu.o: in function `<bustd::utils::effective_user_id::__NoPanic as core::ops::drop::Drop>::drop':
          bustd.4072317c-cgu.0:(.text._ZN84_$LT$bustd..utils..effective_user_id..__NoPanic$u20$as$u20$core..ops..drop..Drop$GT$4drop17hb0dd8c74268564e8E+0x3): undefined reference to `
          
          ERROR[no-panic]: detected panic in function `effective_user_id`
          '
          /usr/bin/ld: /home/gui/bustd/target/release/deps/bustd-29598bf87e19bc74.bustd.4072317c-cgu.0.rcgu.o: in function `<bustd::utils::running_as_sudo::__NoPanic as core::ops::drop::Drop>::drop':
          bustd.4072317c-cgu.0:(.text._ZN82_$LT$bustd..utils..running_as_sudo..__NoPanic$u20$as$u20$core..ops..drop..Drop$GT$4drop17h8cd11b1c65f8f688E+0x3): undefined reference to `
          
          ERROR[no-panic]: detected panic in function `running_as_sudo`
          '
          /usr/bin/ld: /home/gui/bustd/target/release/deps/bustd-29598bf87e19bc74.bustd.4072317c-cgu.0.rcgu.o: in function `<bustd::utils::page_size::__NoPanic as core::ops::drop::Drop>::drop':
          bustd.4072317c-cgu.0:(.text._ZN76_$LT$bustd..utils..page_size..__NoPanic$u20$as$u20$core..ops..drop..Drop$GT$4drop17h3810a645249e8a35E+0x3): undefined reference to `
          
          ERROR[no-panic]: detected panic in function `page_size`
          '
          collect2: error: ld returned 1 exit status
          
  = help: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
  = note: use the `-l` flag to specify native libraries to link
  = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorustc-link-libkindname)

error: could not compile `bustd` due to previous error
```

Any tips? Thanks.

* rustc 1.56.1 (Arch Linux rust 1:1.56.1-3)
* cc (GCC) 11.1.0
* linux 5.10.78-1-lts

**Comment by vrmiguel** [comment]
Something changed in the [`no-panic`](https://github.com/dtolnay/no-panic) macro which introduced this problem :/

I'll try to solve this as soon as possible!

**Comment by vrmiguel** [comment]
Closed by 9c7699626b1dc91f8b48da16897a2190e62edca4   :D

**Comment by guihkx** [comment]
Thanks!
